src/gobexport/requests.py

```diff
@@ -73,6 +73,29 @@ def post(url, json):
     return _exec(requests.post, url=url, headers=_updated_headers(url), json=json, timeout=_REQUEST_TIMEOUT)
 
 
+def handle_streaming_gob_response(func):
+    """Wraps streaming endpoints, adds error handling as implemented by GOB-API
+    GOB-API always returns an empty line on a successful request. If the last line is not an empty line, an
+    APIException is raised.
+
+    :param func:
+    :return:
+    """
+    def wrapper(*args, **kwargs):
+        last_item = None
+        for item in func(*args, **kwargs):
+            last_item = item
+
+            if last_item != b'':
+                yield item
+
+        if last_item != b'':
+            raise APIException(f"Incomplete request received from API. See API logs for more info.")
```

Comment:
Deliberately not checking for message. GOB-Export shouldn't care why this request is incomplete. Keeps the logic simple (only check for empty line). GOB-API is responsible for logging the error.

Reply:
Well, a yield of '' is a valid yield. Why not simple yield item and finally only testing if the last line was ''?

Reply:
Code down the line performs json decodes on the result, which fails on "". The API is never expected to return empty lines anyway (and if so, it's kind of useless), except for the last line.